### DISCUSSION TITLE
Main vocal vs chorus

### DISCUSSION THREAD

**Discussion by dwaipayansinha** [intel/openvino-plugins-ai-audacity#302] [Ideas]
Please add the capability to split the vocals to main vocal and chorus parts.... currently it just gives one vocal track.

**Comment by RyanMetcalfeInt8** [comment]
Thanks for the recommendation!

If someone can recommend an open source AI model that is able to do this, I'd be happy to take a look at potentially porting the feature.

**Comment by RyanMetcalfeInt8** [comment]
Perhaps a good 'speaker source separation' model could be used here. And it's been a while since I've evaluated those, but I haven't been too impressed with the quality produced by the ones that I've found (that had pre-trained models available). 

@MarcoRavich or others may have stumbled upon something better suited.

**Reply by MarcoRavich** [reply]
Hi there, 
as you probably already know the "de facto" standards in AI audio separation field are [demucs](https://github.com/facebookresearch/demucs#readme) by @facebookresearch and [spleeter](https://github.com/deezer/spleeter#readme) by @deezer...
...as always there are many interesting competitors (such as @sigsep's [Open-Unmix](https://sigsep.github.io/open-unmix/) for example) [listed @ HyMPS](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-based.md#separating-) but, if you're looking for "state of the art" ones, I strongly suggest you to check recent international challenges results such as @sdx-workshop's one:
- [Sound Demixing Challenge 2023](https://www.aicrowd.com/challenges/sound-demixing-challenge-2023) - Build A System To Perform Audio Source Separation
- The Sound Demixing Challenge 2023 (summary paper) – [Music Demixing Track](https://transactions.ismir.net/articles/10.5334/tismir.171) or [Cinematic Demixing Track](https://transactions.ismir.net/articles/10.5334/tismir.172)

Some GH-available projects/authors that participated in it:
- [BS-RoFormer](https://github.com/lucidrains/BS-RoFormer#readme) by @lucidrains;
- [MVSEP-MDX23-music-separation-mode](https://github.com/ZFTurbo/MVSEP-MDX23-music-separation-model)l by @ZFTurbo;
- [kmdx-net Music Source Separation](https://github.com/KimberleyJensen/kmdx-net_music-source-separation#readme) by @KimberleyJensen;
- [kuielab](https://github.com/kuielab/sdx23#readme) by @kuielab (@rlaalstjr47 & @junicusmaximus);
- [AIMLESS](https://github.com/aim-qmul/sdx23-aimless#readme) by @aim-qmul (@yoyololicon [& c](https://github.com/aim-qmul/sdx23-aimless/graphs/contributors)).

Last but not least, there are interesting [issue](https://github.com/Anjok07/ultimatevocalremovergui/issues/593)/[discussion](https://github.com/Anjok07/ultimatevocalremovergui/discussions/689) in @Anjok07's [Ultimate Vocal Remover GUI git](https://github.com/Anjok07/ultimatevocalremovergui#readme) where other interesting resources has been shared (such as @TRvlvr's [repositories](https://github.com/TRvlvr?tab=repositories) for example).

Hope that helps.

**Reply by Brodvd** [reply]
Small clarification: 
I took advantage of this discussion to discuss a model that has capacity for separation of sound events wider, such as Audio Sep, perhaps in the specific case of  @dwaipayansinha  there is a need for a model on purpose, I do not know.
However we can choose more models.

**Comment by Brodvd** [comment]
I'm not sure but maybe Audio Sep can funtion? https://github.com/Audio-AGI/AudioSep#readme

**Reply by RyanMetcalfeInt8** [reply]
> I'm not sure but maybe Audio Sep can funtion? https://github.com/Audio-AGI/AudioSep#readme

Whoa, nice -- that is super cool -- thanks for pointing it out. I haven't stumbled on this one yet.

**Reply by Brodvd** [reply]
You can tried It here: https://huggingface.co/spaces/Suniilkumaar/AudioSep

**Reply by RyanMetcalfeInt8** [reply]
Although reading through some of the issues filed in that GitHub project, it seems like many are not too impressed with the results in 'real world' use cases. So the concept for this one is great, but if the pre-trained model doesn't work very well, I'm going to just avoid it for now.

**Reply by RyanMetcalfeInt8** [reply]
> You can tried It here: https://huggingface.co/spaces/Suniilkumaar/AudioSep

Thanks -- maybe I'll also give it a shot when I get a chance.

**Reply by Brodvd** [reply]
@RyanMetcalfeInt8  I think that Is better that you try. Also only for use It for voice and instruments Is good

**Comment by Brodvd** [comment]
@RyanMetcalfeInt8  Sorry if I fill you with notifications. I ask for a moment of patience. I share a little test I did with Audio Sep: 

![image](https://github.com/user-attachments/assets/9689b5cc-8843-42f3-98b1-f3d0d5e71558)

below the output of Audio Sep:

https://github.com/user-attachments/assets/caeb4646-97e2-49a7-9010-67def6d9f89e


As you said the idea of this model is fantastic and (I think) it’s the only one that does this. 

Obviously it is impossible that a model like this which must associate text with audio characteristics (with separation) totally covers all the sounds, just think of the thousands of ambient sounds, but if you use it consciously this model (as in this case) is absolutely great performance. 

So I ask you not to discard Audio Sep at all but keep it very much in mind because if you use Audio Sep to separate musical instruments or voices, you don’t have to resort to stems.


Think about it

**Reply by RyanMetcalfeInt8** [reply]
wow, very cool, thanks for sharing -- I'll definitely give this a try. It would be a very cool feature to include, and if it works this well (as you said, for specific cases) I think it would come in very handy!

**Reply by Brodvd** [reply]
Thanks a lot :)
Sorry if I haven't others examples. I haven't Anaconda.

**Comment by Brodvd** [comment]
Update : I tried to run Audio Sep but there are conflict with the libreries, the same said [here](https://github.com/Audio-AGI/AudioSep/issues/37). 
Anyway Audio Sep can be run with [pinokio](https://pinokio.computer/), like said [here](https://github.com/Audio-AGI/AudioSep/issues/44) , but I don't know.
I used Miniconda 3.12 and python 3.9, maybe it can be funtion with Miniconda 3.9.
